### ISSUE TITLE
Could not control access plan on REL95_1_1_8 in one use case

### ISSUE THREAD

**Issue by higuchi-daisuke** [ossc-db/pg_hint_plan#30]
Hi, 

I found the bug on REL95_1_1_8. 
I tried to force MergeJoin, but I could not control access plan in special use case. 

Platform information:
OS Version : RHEL 7.4
PostgreSQL version : 9.5.16
pg_hint_plan version : REL95_1_1_8

# How to reproduce
Initial setup as follows:
```
CREATE EXTENSION pg_hint_plan;
CREATE EXTENSION pg_dbms_stats;
CREATE SCHEMA S1;
CREATE TABLE S1.T1(C1 int);
CREATE TABLE S1.T2(C1 int);
CREATE UNIQUE INDEX ix1T1 ON S1.T1(C1);
CREATE UNIQUE INDEX ix1T2 ON S1.T2(C1);
INSERT INTO S1.T1(C1) SELECT generate_series(1,10);
INSERT INTO S1.T2(C1) SELECT generate_series(1,10);
ANALYZE;
```

After this, execute "psql -f test.sql". test.sql has following SQLs:
```
EXPLAIN SELECT T1.C1,T2.C1 FROM S1.T1,S1.T2 WHERE T1.C1=T2.C1;
ANALYZE;
SELECT dbms_stats.lock_table_stats('s1.t1');
EXPLAIN SELECT /*+ MergeJoin(t1 t2) */ T1.C1,T2.C1 FROM S1.T1,S1.T2 WHERE T1.C1=T2.C1;
SELECT dbms_stats.unlock_table_stats('s1.t1');
```

Even if MergeJoin is specified in hint clause, HashJoin is selected. 

Note: 
As far as I saw, this problem occurs only when I use "psql -f xxx". 
This problem does not occur if I execute SQLs interactively on psql terminal. 

# Investigation

After adding [this commit](https://github.com/ossc-db/pg_hint_plan/commit/000c325874cc07eb944447d046a122dcc8642520) to REL95_1_1_8, this problem is solved. 
The commit log shows "9.5 is not affected", but I think this should be back-patch to 9.5 too. 

Regards, 
Daisuke Higuchi

**Comment by horiguti** [comment]
The repor doesn't work for me, with any ordering of the two extensions in shared_preload_libraries and both -f or not.  Do you suppose some other conditions for the behavior?

**Comment by higuchi-daisuke** [comment]
Hi, I'm sorry for confusing you. 
I could not reproduce this issue even if I follow the procedure in my first post.. 
So, I write the simple procedure to reproduce. I use pg_hint_plan REL95_1_1_8 and PostgreSQL 9.5.21. 

1. Initial setup by following my first post.
2. Execute following command:
    psql -c "EXPLAIN SELECT /*+ MergeJoin(t1 t2) */ T1.C1,T2.C1 FROM S1.T1,S1.T2 WHERE T1.C1=T2.C1;"

I confirm not Merge Join but Hash Join from explain information of 2. 

[ A little investigation ]
When new session starts, pg_dbms_stats needs to search "locked statistics" tables, I think. 
If so, this behavior is similar to the following comment of commit 000c325.
```
Hints lose effect when pg_dbms_stats searches "locked statistics" tables 
while planning the target query.
```

**Comment by horiguti** [comment]
OK, I confirmed that when both pg_hint_plan and pg_dbms_stats are preloaded, running the given query as the first query in a session doesn't honor the hint.  Some digging on the repository told me that a change committed fife years ago (9.4 era) have broken the previous behavior.
Fixed it. Patched from PG94 up to master (PG12).

https://github.com/ossc-db/pg_hint_plan/commit/4e63a22d0aeb67b03f05ddc689f25e8a0b27ef91
